lib/ansible/module_utils/network/ftd/configuration.py

```diff
@@ -28,86 +30,361 @@
+                        existing_obj)
 
-    def delete_object(self, url_path, path_params):
+        raise e
+
+    def _find_get_list_operation(self, model_name):
+        operations = self.get_operation_specs_by_model_name(model_name) or {}
+        return next((
+            op for op, op_spec in operations.items()
+            if self._operation_checker.is_get_list_operation(op, op_spec)), None)
+
+    def _find_get_operation(self, model_name):
+        operations = self.get_operation_specs_by_model_name(model_name) or {}
+        return next((
+            op for op, op_spec in operations.items()
+            if self._operation_checker.is_get_operation(op, op_spec)), None)
+
+    def delete_object(self, operation_name, params):
         def is_invalid_uuid_error(err):
             return err.code == UNPROCESSABLE_ENTITY_STATUS and INVALID_UUID_ERROR_MESSAGE in str(err)
 
         try:
-            return self.send_request(url_path=url_path, http_method=HTTPMethod.DELETE, path_params=path_params)
+            return self.send_general_request(operation_name, params)
         except FtdServerError as e:
             if is_invalid_uuid_error(e):
                 return {'status': 'Referenced object does not exist'}
             else:
                 raise e
 
-    def edit_object(self, url_path, body_params, path_params=None, query_params=None):
-        existing_object = self.send_request(url_path=url_path, http_method=HTTPMethod.GET, path_params=path_params)
+    def edit_object(self, operation_name, params):
+        data, dummy, path_params = _get_user_params(params)
 
-        if not existing_object:
-            raise FtdConfigurationError('Referenced object does not exist')
-        elif equal_objects(existing_object, body_params):
-            return existing_object
-        else:
-            return self.send_request(url_path=url_path, http_method=HTTPMethod.PUT, body_params=body_params,
-                                     path_params=path_params, query_params=query_params)
+        model_name = self.get_operation_spec(operation_name)[OperationField.MODEL_NAME]
+        get_operation = self._find_get_operation(model_name)
+
+        if get_operation:
+            existing_object = self.send_general_request(get_operation, {ParamName.PATH_PARAMS: path_params})
+            if not existing_object:
+                raise FtdConfigurationError('Referenced object does not exist')
+            elif equal_objects(existing_object, data):
+                return existing_object
+
+        return self.send_general_request(operation_name, params)
 
-    def send_request(self, url_path, http_method, body_params=None, path_params=None, query_params=None):
+    def send_general_request(self, operation_name, params):
+        def stop_if_check_mode():
+            if self._check_mode:
+                raise CheckModeException()
+
+        self.validate_params(operation_name, params)
+        stop_if_check_mode()
```

Comment:
Not sure what is the advantage of defining non repeating functionality of stop_if_check_mode inside function

Reply:
I think we used to use it in multiple places, but now it's redundant indeed, will remove.